modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Fields.tsx

```diff
@@ -68,16 +76,54 @@ export default function Fields({
 
 	useEffect(() => {
 		const makeFetch = async () => {
-			const objectDefinition = await API.getObjectDefinitionByExternalReferenceCode(
+			setLoad(true);
+
+			const objectDefinitionResponse = await API.getObjectDefinitionByExternalReferenceCode(
 				objectDefinitionExternalReferenceCode
 			);
 
-			setCreationLanguageId(objectDefinition.defaultLanguageId);
+			const url = createResourceURL(baseResourceURL, {
+				objectDefinitionId: objectDefinitionResponse.id,
+				p_p_resource_id: '/object_definitions/get_object_field_types',
+			}).href;
+
+			const objectFieldTypesResponse = await fetch(url, {
+				method: 'GET',
+			});
+
+			const {
+				objectFieldTypes: newObjectFieldTypes,
+			} = (await objectFieldTypesResponse.json()) as {
+				objectFieldTypes: ObjectFieldType[];
+			};
+
+			setCreationLanguageId(objectDefinitionResponse.defaultLanguageId);
+			setObjectFieldTypes(newObjectFieldTypes);
+
+			setTimeout(() => {
+				setLoad(false);
+			}, 200);
 		};
 
 		makeFetch();
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [objectDefinitionExternalReferenceCode]);
 
+	function objectFieldBusinessTypeDataRenderer({
+		itemData,
+	}: {
+		itemData: ItemData;
+	}) {
+		const businessTypeMap = new Map<string, ObjectFieldType>();
+
+		objectFieldTypes.forEach((objectFieldType) => {
+			businessTypeMap.set(objectFieldType.businessType, objectFieldType);
+		});
```

Comment:
Hey @igor-franca, in this case, for each field we will create a `map`, do a `forEach` to define the map and `get` an item in a map. We can optimize this entire process by doing it once, in the `useEffect` above and configuring `setObjectFieldTypes` with this map. This `objectFieldBusinessTypeDataRenderer` function will only be used to obtain the value;

```tsx
function objectFieldBusinessTypeDataRenderer({
   itemData,
}: ObjectFieldItemData) {
   return objectFieldTypes?.get(itemData.businessType)?.label;
}
```